tldextract/tldextract.py

```diff
@@ -187,8 +187,15 @@ def __call__(self, url):
             .rstrip(".")
 
         is_punycode = netloc.startswith('xn--') or '.xn--' in netloc
+        # Some IDN urls might generate an error
         if is_punycode:
-            netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            try:
+                netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            except UnicodeError:
+                LOG.warn('UnicodeError when decoding netloc ' + netloc)
```

Comment:
One final nitpick: is this log necessary? This hunk is `tldextract`'s attempt to interpret the input as punycode. If `tldextract` guessed wrong, that's on the library, not the user. The user doesn't need to know. Similar to the `try`/`except` in lines 207-214.